adaptive_memory_v4.0.py

```
@@ -0,0 +1,1982 @@
+
+    async def load_embedding_persistent(self, user_id: str, memory_id: str) -> Optional[np.ndarray]:
+        """Load a stored embedding from the persistent JSON file."""
+        try:
+            cache_dir = "/app/backend/data/cache/embeddings"
+            cache_file = os.path.join(cache_dir, f"{user_id}_embeddings.json")
+            
+            if not os.path.exists(cache_file):
+                return None
+            
+            # Load cache
+            with open(cache_file, 'r') as f:
```

Ruff (0.14.10)

Async functions should not open files with blocking methods like `open`

(ASYNC230)